sdk/lib/drivers/ip/transport/tcp/accept.c

```diff
@@ -77,10 +77,18 @@ NTSTATUS TCPListen(PCONNECTION_ENDPOINT Connection, UINT Backlog)
             if (NT_SUCCESS(Status))
             {
                 /* Allocate the port in the port bitmap */
-                Connection->AddressFile->Port = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
-
-                /* This should never fail */
-                ASSERT(Connection->AddressFile->Port != 0xFFFF);
+                UINT nPort = TCPAllocatePort(LocalAddress.Address[0].Address[0].sin_port);
+
+                if (nPort == (UINT)-1)
```

Comment:
If this should never fail, then an assert on this would be a good idea?
Otherwise, please elaborate on why it's allowed to fail now.

Reply:
Now that we properly check against (UINT)-1 and no longer 0xFFFF, we could indeed put the ASSERT back. But then i'd strongly advise to re-run the test with the same app provided in the ticket.